integration/management-api-tests/src/test/java/org/wso2/micro/integrator/api/ConfigsResourceTestCase.java

```diff
@@ -0,0 +1,290 @@
+                    logLine.contains("ROUND-TRIP LATENCY") || logLine.contains("Thread switch latency"));
+        }
+
+        sendGetRequest(offset, true);
+
+        while ((logLine = bufferedReader.readLine()) != null) {
+            Assert.assertTrue(logLine.contains("HTTP State Transition") ||
+                    logLine.contains("ROUND-TRIP LATENCY") || logLine.contains("Thread switch latency"));
+        }
+    }
+
+    /**
+     * Start a new server.
+     *
+     * @throws IOException
+     * @throws AutomationFrameworkException
+     */
+    private void startNewServer(boolean enableCorrelationLogs) throws IOException, AutomationFrameworkException {
+
+        HashMap<String, String> startupParameterMap = new HashMap<>();
+        startupParameterMap.put("-DportOffset", String.valueOf(offset));
+        startupParameterMap.put("-DenableCorrelationLogs", String.valueOf(enableCorrelationLogs));
+        startupParameterMap.put("startupScript", "micro-integrator");
+
+        server = new CarbonTestServerManager(context, System.getProperty("carbon.zip"), startupParameterMap);
+        server.startServer();
+        Awaitility.await().pollInterval(50, TimeUnit.MILLISECONDS).atMost(DEFAULT_TIMEOUT, TimeUnit.SECONDS).
+                until(isManagementApiAvailable());
+        carbonHome = server.getCarbonHome();
+        Assert.assertNotNull(carbonHome);
+    }
+
+    private void stopServer() throws AutomationFrameworkException {
+
+        server.stopServer();
+    }
+
+    private void sendGetRequest(int offset, boolean correlationConfigStatus) throws IOException {
+
+        String accessToken = TokenUtil.getAccessToken(hostName, offset);
+        Assert.assertNotNull(accessToken);
+
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Accept", "application/json");
+        headers.put("Authorization", "Bearer " + accessToken);
+
+        String endpoint = "https://" + hostName + ":" + (DEFAULT_INTERNAL_API_HTTPS_PORT + offset) + "/management/"
+                + "configs?configName=correlation";
+
+        SimpleHttpClient client = new SimpleHttpClient();
+
+        HttpResponse response = client.doGet(endpoint, headers);
+        String responsePayload = client.getResponsePayload(response);
+        Assert.assertEquals(response.getStatusLine().getStatusCode(), 200);
+        JSONObject jsonResponse = new JSONObject(responsePayload);
+        Assert.assertTrue(jsonResponse.has("configName"));
+        Assert.assertEquals(jsonResponse.get("configName"), "correlation");
+        Assert.assertTrue(jsonResponse.has("configs"));
+        JSONObject responseConfigs = (JSONObject) jsonResponse.get("configs");
+        Assert.assertTrue(responseConfigs.has("enabled"));
+        Assert.assertEquals(responseConfigs.get("enabled"), correlationConfigStatus);
+    }
+
+    private void sendPutRequest(int offset, boolean correlationConfigEnabled) throws IOException {
```

Comment:
Something like `updateCorrelationLogState` will be more meaningful. WDYT?

Reply:
Yeah. It seems more meaningful